api/agent/pure_runner.go

```diff
@@ -626,11 +641,112 @@ DataLoop:
 	return state.waitError()
 }
 
+// Runs a status call using status image with baked in parameters.
+func (pr *pureRunner) runStatusCall(ctx context.Context, result *runner.RunnerStatus) {
+
+	log := common.Logger(ctx)
+	start := time.Now()
+
+	// construct call
+	var c models.Call
+
+	// Most of these arguments are baked in. We might want to make this
+	// more configurable.
+	c.ID = id.New().String()
+	c.Path = "/"
+	c.Image = pr.statusImageName
+	c.Type = "sync"
+	c.Format = "json"
+	c.TmpFsSize = 0
+	c.Memory = 0
+	c.CPUs = models.MilliCPUs(0)
+	c.URL = "/"
+	c.Method = "GET"
+	c.CreatedAt = common.DateTime(start)
+	c.Config = make(models.Config)
+	c.Config["FN_FORMAT"] = c.Format
+	c.Payload = "{}"
+
+	// Try to fetch ctx deadline
+	deadline, ok := ctx.Deadline()
+	if ok {
+		c.Timeout = int32(math.Min(float64(deadline.Sub(start)/time.Second), float64(1)))
+	} else {
+		c.Timeout = 10
+	}
+	c.IdleTimeout = 1
+
+	// TODO: reliably shutdown this container after executing one request.
+
+	log.Debugf("Running status call with id=%v timeout=%v image=%v", c.ID, c.Timeout, c.Image)
+
+	recorder := httptest.NewRecorder()
+	player := ioutil.NopCloser(strings.NewReader(c.Payload))
+
+	agent_call, err := pr.a.GetCall(FromModelAndInput(&c, player),
+		WithWriter(recorder),
+		WithContext(ctx),
+	)
+
+	if err == nil {
+		var mcall *call
+		mcall = agent_call.(*call)
+		err = pr.a.Submit(mcall)
+	}
+
+	resp := recorder.Result()
+
+	if err != nil {
+		result.ErrorCode = int32(models.GetAPIErrorCode(err))
+		result.ErrorStr = err.Error()
+		result.Failed = true
+	} else if resp.StatusCode >= http.StatusBadRequest {
+		result.ErrorCode = int32(resp.StatusCode)
+		result.Failed = true
+	}
+
+	// These timestamps are related. To avoid confusion
+	// and for robustness, nested if stmts below.
+	if !time.Time(c.CreatedAt).IsZero() {
+		result.CreatedAt = c.CreatedAt.String()
+
+		if !time.Time(c.StartedAt).IsZero() {
+			result.StartedAt = c.StartedAt.String()
+
+			if !time.Time(c.CompletedAt).IsZero() {
+				result.CompletedAt = c.CompletedAt.String()
+			} else {
+				// IMPORTANT: We punch this in ourselves.
+				// This is because call.End() is executed asynchronously.
+				result.CompletedAt = common.DateTime(time.Now()).String()
+			}
+		}
+	}
+
+	// Status images should not output excessive data since we echo the
+	// data back to caller.
+	body, _ := ioutil.ReadAll(resp.Body)
+
```

Comment:
Worth adding a comment that we're truncating this to the last 64 bytes of the response body?

Reply:
Sounds good.